File: JavaScript/CreditCardValidator.js
function getCreditCardType(cardNumber) {
    if (cardNumber.startsWith('4')) {
        return 'Visa Card';
    } else if (cardNumber.startsWith('5')) {
        return 'Master Card';
    } else if (cardNumber.startsWith('37')) {
        return 'American Express Cards';
    } else if (cardNumber.startsWith('6')) {
        return 'Discover cards';
    } else {
        return 'Invalid Input';
    }
}

function getCreditCardDigitLength(cardNumber) {
    return cardNumber.length;
}

function getCreditCardValidityStatus(cardNumber) {
    let sum = 0;
    let alternate = false;

    for (let i = cardNumber.length - 1; i >= 0; i--) {
        let digit = parseInt(cardNumber[i]);

        if (alternate) {
            digit *= 2;
            if (digit > 9) {
                digit -= 9; // equivalent to summing the two digits
            }
        }

        sum += digit;
        alternate = !alternate;
    }

    return sum % 10 === 0 ? 'Valid' : 'Invalid';
}


let userInput = prompt('Enter a credit card number: ').replace(/[^0-9]/g,);
console.log(`Card Type: ${getCreditCardType(userInput)}`);
console.log(`Card Digit Length: ${getCreditCardDigitLength(userInput)}`);
console.log(`Card Validity Status: ${getCreditCardValidityStatus(userInput)}`);

File: Java/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {

    public static String getCardType(String number) {
        if (number.startsWith("4")) {
            return "Visa Card";
        } else if (number.startsWith("5")) {
            return "Master Card";
        } else if (number.startsWith("37")) {
            return "American Express Card";
        } else if (number.startsWith("6")) {
            return "Discover Card";
        } else {
            return "Invalid Input";
        }
    }

    public static int getCardLength(String number) {
        return number.length();
    }

    public static String getCardValidity(String number) {
        int sum = 0;
        boolean alternate = false;

        for (int i = number.length() - 1; i >= 0; i--) {
            int digit = Integer.parseInt(String.valueOf(number.charAt(i)));

            if (alternate) {
                digit *= 2;
                if (digit > 9) {
                    digit -= 9;
                }
            }

            sum += digit;
            alternate = !alternate;
        }

        return (sum % 10 == 0) ? "Valid" : "Invalid";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a credit card number: ");
        String input = scanner.nextLine();

        System.out.println("Card Type: " + getCardType(input));
        System.out.println("Card Digit Length: " + getCardLength(input));
        System.out.println("Card Validity Status: " + getCardValidity(input));

     }
}